server/src/main/java/org/apache/cassandra/sidecar/restore/RestoreJobDiscoverer.java

```diff
@@ -154,11 +163,33 @@ public long delay()
     @Override
     public void execute(Promise<Void> promise)
     {
-        executeInternal();
+        tryExecute();
         promise.tryComplete();
     }
 
-    public void executeInternal()
+    /**
+     * Try to execute the job discovery task in the blocking way. It returns immediately, if another thread is executing already.
+     * The method can be invoked by external call-sites.
+     */
+    public void tryExecute()
```

Comment:
NIT: maybe a more descriptive name is suitable here since it is in the public API
```suggestion
    public void runDiscoverer()
```